plugins/woocommerce/tests/legacy/unit-tests/customer/crud.php

```diff
@@ -399,6 +399,18 @@ public function test_customer_set_shipping_address_to_base() {
 		$this->assertEmpty( $customer->get_shipping_city() );
 	}
 
+	/**
+	 * Test getting the default location with no default address set but specific countries allowed.
+	 */
+	public function test_default_location_with_specific_allowed_countries() {
+		delete_option( 'woocommerce_default_customer_address' );
+		update_option( 'woocommerce_allowed_countries', 'specific' );
+		update_option( 'woocommerce_specific_allowed_countries', array( 'DE', 'AT', 'CH' ) );
```

Comment:
🤔 There's an assumption about a default value here in the test that I think might be worth making more explicit in case that default changes later. The `$location` value that eventually gets checked against the list of allowed countries is initially populated by the `woocommerce_default_country` option, with a fallback to `US:CA`. So this test works because `US` isn't in the allowed array specified. But I think it would be safer to explicitly set a value on `woocommerce_default_country` here that isn't in the array so that we can be sure the test will always work right... what do you think?

Reply:
Thanks @coreymckrill, those are some great ideas -- given that these things don't seem to be covered by any other tests yet I'll add a few different cases. 👍